### PR TITLE
feat: Adds eth/avl head endpoint cache control params

### PR DESCRIPTION
**Pull request by moraesjeremias**
# Problem Statement

During cache test/implementation we found some hardcoded cache control headers that were blocking cache invalidation on Proxy side.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -477,7 +479,13 @@ async fn get_eth_head(State(state): State<Arc<AppState>>) -> impl IntoResponse {
                             let now = Utc::now().timestamp() as u64;
                             (
                                 StatusCode::OK,
-                                [("Cache-Control", "public, max-age=7200, must-revalidate")],
+                                [(
+                                    "Cache-Control",
+                                    format!(
+                                        "public, max-age={}, must-revalidate",
+                                        state.eth_head_response_cache
+                                    ),
```
Comment (moraesjeremias) [inline src/main.rs:487]: Why it didn't work before? 🤔

File: .env.example [hunk 2]
```diff
@@ -6,5 +6,7 @@ VECTORX_CONTRACT_ADDRESS=0xbc281367e1F2dB1c3e92255AA2F040B1c642ec75
 BRIDGE_CONTRACT_ADDRESS=0x1369a4c9391cf90d393b40faead521b0f7019dc5
 ETHEREUM_CLIENT_URL=https://ethereum-sepolia.publicnode.com
 BEACONCHAIN_URL=https://sepolia.beaconcha.in/api/v1/slot
+ETH_HEAD_RESPONSE_CACHE=240
```
Comment (QEDK) [inline .env.example:9]:
```suggestion
ETH_HEAD_CACHE_MAXAGE=240
AVL_HEAD_CACHE_MAXAGE=600
```
Better to be a bit more verbose so it's clear what this variable is doing.

File: src/main.rs [hunk 3]
```diff
@@ -490,13 +498,23 @@ async fn get_eth_head(State(state): State<Arc<AppState>>) -> impl IntoResponse {
                             if err.to_string().ends_with("status code: 429") {
                                 (
                                     StatusCode::TOO_MANY_REQUESTS,
-                                    [("Cache-Control", "max-age=300, must-revalidate")],
+                                    [
+                                        (
+                                            "Cache-Control",
+                                            "max-age=300, must-revalidate".to_string(),
```
Comment (QEDK) [inline src/main.rs:504]: Doesn't work without `to_string()`?

**Review by QEDK — CHANGES_REQUESTED** [review]
Couple of things.

File: src/main.rs (hunk 1 above)
Comment (0xSasaPrsic) [reply src/main.rs:508]: different types were returned, in this case, it is `&str` and `String` and in the other place was `&str` and `&str`

File: src/main.rs (hunk 3 above)
Comment (0xSasaPrsic) [reply src/main.rs]: no, because of `format!` that returns `String` and when converting to `to_str` needs a different lifetime

**Review by QEDK — APPROVED** [review]
LGTM.

**Pull request merged**